erpnext/stock/doctype/delivery_note/delivery_note.py

```diff
@@ -209,6 +208,11 @@ def on_submit(self):
 		elif self.issue_credit_note:
 			self.make_return_invoice()
 
+		if self.delivered:
+			frappe.db.set_value("Delivery Note", self.name, "status", "Delivered")
+		else:
+			frappe.db.set_value("Delivery Note", self.name, "status", "To Deliver")
```

Comment:
Do this on `before_submit` instead:

```suggestion
		self.status = "Delivered" if self.delivered else "To Deliver"
```